pyigm/clustering/clustering_field.py

```diff
@@ -0,0 +1,241 @@
+""" Module containing the ClusteringField class
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+from scipy.ndimage import gaussian_filter as gf
+
+from pyigm.field.igmfield import IgmGalaxyField
+from pyigm.clustering.xcorr_utils import random_gal, auto_pairs_rt, cross_pairs_rt
+
+
+class ClusteringField(IgmGalaxyField):
+    """The Field class is meant to contain information from a given
+    galaxy field (with one or more QSO sightlines containing IGM
+    information).  It sub-classes on IGMGalaxyField which is expected
+    to hold *all* of the galaxies for the field of interest.
+
+    Input parameters:
+    ---
+    radec : coordinate
+    absreal:   catalog of real absorption lines (numpy rec array). It has to have
+               dtype.names RA,DEC,ZABS,LOGN,B
+    galreal:   catalog of real galaxies (numpy rec array). It has to have
+               dtype.names RA, DEC, ZGAL, MAG
+    wa:        numpy array with QSO spectral coverage (wavelength)
+    fl:        numpy aray with the QSO spectrum flux it can be normalized or not)
+    er:        numpy array with the QSO spectrum error (it can be normalized
+               or not, but has to be consistent with fl)
+    R:         resolution of the QSO spectrum spectrograph
+    Ngal_rand: number of random galaxies per real one that will be created.
+    Nabs_rand: number of random absorbers per real one that will be created.
+    proper:    calculates everything in physical Mpc rather than co-moving (Boolean).
+
```

Comment:
wrapped : bool
Whether to wrap the pair counts along the line of sight to increase counts (if the signal is isotropic this is justified).

Reply:
added